### PR TITLE
SEND_HELP

### PR DESCRIPTION
**Pull request by barnabwhy**

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -174,4 +180,90 @@ module.exports = {
...
+            return cat.toLowerCase().startsWith(opts.getText('page').toLowerCase())
+        })
+
+        if (typeof catPage == 'undefined') {
+            source.channel.send({
+                embed: {
+                    title: 'Categories',
+                    footer: {
+                        icon_url: state.avatar,
+                        text: 'WAHtson Help | Categories',
```
Comment (nanaian) [inline src/actions.js:216]: Drop "WAHtson" from this string

File: src/actions.js [hunk 2]
```diff
@@ -174,4 +180,90 @@ module.exports = {
...
+                    ],
+                },
+            })
+        } else {
+            source.channel.send({
+                embed: {
+                    title: catPage,
+                    footer: {
+                        icon_url: state.avatar,
+                        text: `WAHtson Help | ${catPage}`,
```
Comment (nanaian) [inline src/actions.js:247]: & here

File: src/actions.js [hunk 3]
```diff
@@ -174,4 +180,90 @@ module.exports = {
...
+                        icon_url: state.avatar,
+                        text: 'WAHtson Help | Categories',
+                    },
+                    author: {
+                        name: 'Help',
+                        icon_url: state.avatar,
+                    },
+                    fields: [
+                        categories.map(cat => {
+                            var catCmds = commands.filter(cmd => {
```
Comment (nanaian) [inline src/actions.js:224]: const

File: src/actions.js [hunk 4]
```diff
@@ -212,22 +212,20 @@ module.exports = {
...
+                            const catCmds = commands.filter(cmd => {
                                 return cmd.category == cat
                             })
-                            var catCmdsLength = catCmds.length
-                            catCmds.length = 3
-                            if (catCmdsLength > 3)
-                                catCmds[3] = { usage: `**${catCmdsLength} more commands**` }
+                            if (catCmds.length > 3)
+                                catCmds[3] = { usage: `**${catCmds.length - 3} more commands**` }
+                                catCmds.length = 4
```
Comment (nanaian) [inline src/actions.js:228]: What's going on here? You're _setting_ the length? Why?

File: src/util.js [hunk 5]
```diff
@@ -150,6 +150,13 @@ function safeToString(x) {
     }
 }
 
+const uniqueArray = a => {
+    a = a.filter(function (item, index) {
```
Comment (nanaian) [inline src/util.js:154]: Would be better for this to be an arrow function

File: src/actions.js (hunk 4 above)
Comment (barnabwhy) [reply src/actions.js:228]: Limiting the length so that additional commands don't show

File: src/util.js [hunk 6]
```diff
@@ -150,6 +150,13 @@ function safeToString(x) {
     }
 }
 
+const uniqueArray = a => {
+    a = a.filter((item, index) => {
+        return a.indexOf(item) == index
+    })
+    return a
+}
```
Comment (nanaian) [inline src/util.js:158]:
```suggestion
const uniqueArray = a => a.filter((item, index) => (
    a.indexOf(item) == index
))
```

**Pull request merged**